### PR TITLE
optimization: Request cached property

### PR DESCRIPTION
**Pull request by EliAndrewC**
We currently have two types of properties:

1) Regular old ``@property``.  These get invoked every time.

2) ``@cached_property`` which is invoked once and then never again for the lifetime of an object.

We found that we needed a third option, which is ``@request_cached_property``, which caches for the duration of a request.

I've added ``@request_cached_property`` to Sideboard, which means that https://github.com/magfest/sideboard/pull/20 MUST be merged BEFORE this is merged, or else this branch will be invalid.

- [ ] This box is checked if someone verified this actually still works correctly before merging

### CODE REVIEW

File: uber/config.py [hunk 1]
```diff
@@ -273,7 +271,6 @@ def CURRENT_ADMIN(self):
     def HTTP_METHOD(self):
         return cherrypy.request.method
 
-    @property
     def SUPPORTER_COUNT(self):
```
Comment (binary1230) [inline uber/config.py]: i feel like even though we can't use the decorator form here, we should probably mark these with a comment like:
```
# IMPORTANT - Cherrypy Request-specific caching in place here, see request_cached_property()
```

Comment (kitsuta) [inline uber/config.py]: +1 a bit omhbity

Comment (EliAndrewC) [inline uber/config.py]: Actually the more I think about it, the more I believe that ``@request_cached_property`` should just go into Sideboard itself alongside ``@cached_property``.  Then we can just use it directly here and don't need a weird comment.

I'll put together a PR in the Sideboard repo tonight that does that and update this PR to make use of it.

File: uber/config.py [hunk 2]
```diff
@@ -503,7 +507,7 @@ def _is_intstr(s):
 c.MERCH_SHIRT_OPTS = [(c.SIZE_UNKNOWN, 'select a size')] + list(c.PREREG_SHIRT_OPTS)
 c.DONATION_TIER_OPTS = [(amt, '+ ${}: {}'.format(amt, desc) if amt else desc) for amt, desc in c.DONATION_TIER_OPTS]
 
-c.DONATION_TIER_DESCRIPTIONS = _config.get('donation_tier_descriptions', [])
+c.DONATION_TIER_DESCRIPTIONS = _config.get('donation_tier_descriptions', {})
```
Comment (EliAndrewC) [inline uber/config.py]: This is an unrelated fix.  If you look at the line below this one, we're expected ``c.DONATION_TIER_DESCRIPTIONS`` to always be a dict, so in my local Vagrant it was blowing up because I didn't have it defined.

**Comment by EliAndrewC** [comment]
I've verified that this works in a local Python REPL but haven't had a chance to try pulling/merging on the staging server and testing there.

File: uber/config.py (hunk 2 above)
Comment (binary1230) [inline uber/config.py]: nice catch

**Comment by binary1230** [comment]
i'll merge this into our staging branch, optimization-master

**Comment by binary1230** [comment]
will merge and re-run the stress tests
